### PR TITLE
test(perf): grow shrink cluster 3 nodes in parallel

### PR DESCRIPTION
**Pull request by soyacz**
We want to verify how fast we can double cluster capacity and shrink it back.

### Testing

- [ ] - draft testing here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/scylla-master-perf-regression-latency-650gb-grow-shrink/

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: performance_regression_test.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def run_workload(self, stress_cmd, nemesis=False, sub_type=None):
         stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, stress_num=1, stats_aggregate_cmds=False)
         if nemesis:
             interval = self.params.get('nemesis_interval')
-            time.sleep(interval * 60)  # Sleeping one interval (in minutes) before starting the nemesis
+            time.sleep(5 * 60)  # Sleeping one interval (in minutes) before starting the nemesis
```
Comment (fruch) [inline performance_regression_test.py:333]: This is just for testing ? right ?

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -4013,6 +4041,14 @@ def disrupt_grow_shrink_cluster(self):
         self._grow_cluster(rack=None)
         self._shrink_cluster(rack=None)
 
+    def disrupt_grow_shrink_cluster_parallel(self):
```
Comment (fruch) [inline sdcm/nemesis.py:4044]: Later we might want to use the same nemesis 

But have it configurable based on scylla capabilities and SCT configuration

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -1262,6 +1262,30 @@ def _add_and_init_new_cluster_node(self, old_node_ip=None, host_id=None,
...
+        self.log.info("Adding %s new nodes to cluster...", count)
+        InfoEvent(message=f'StartEvent - Adding {count} new nodes to cluster').publish()
+        new_nodes = self.cluster.add_nodes(
+            count=count, dc_idx=self.target_node.dc_idx, enable_auto_bootstrap=True, rack=rack)
+        self.monitoring_set.reconfigure_scylla_monitoring()
+
+        try:
+            with adaptive_timeout(Operations.NEW_NODE, node=self.cluster.nodes[0], timeout=timeout):
+                self.cluster.wait_for_init(node_list=new_nodes, timeout=timeout, check_node_health=False)
+            self.cluster.set_seeds()
```
Comment (fruch) [inline sdcm/nemesis.py:1275]: Messing with the seed isn't really needed anymore, just before starting a node.

File: performance_regression_test.py (hunk 1 above)
Comment (soyacz) [reply performance_regression_test.py:333]: I try to make it working ASAP, so I took some shortcuts here and there. Later I'll remove them.

File: sdcm/nemesis.py (hunk 3 above)
Comment (soyacz) [reply sdcm/nemesis.py:1275]: ok, I'll remove it. See I trimmed original procedure (like setting nemesis target) and I'm not sure of it yet.

**Comment by soyacz** [comment]
btw. I'm taking another shortcut - disabled multi rack scenario, as adding multirack support would take more time to implement.

File: sdcm/nemesis.py [hunk 4]
```diff
@@ -4013,6 +4041,14 @@ def disrupt_grow_shrink_cluster(self):
         self._grow_cluster(rack=None)
         self._shrink_cluster(rack=None)
 
+    def disrupt_grow_shrink_cluster_parallel(self):
+        sleep_time_between_ops = self.cluster.params.get('nemesis_sequence_sleep_between_ops')
+        if not self.has_steady_run and sleep_time_between_ops:
+            self.steady_state_latency()
+            self.has_steady_run = True
+        self._grow_cluster_parallel(rack=None)
+        self._shrink_cluster(rack=None)
```
Comment (roydahan) [inline sdcm/nemesis.py:4050]: We need here the part that wait for the balanced cluster and then double the load.
During doubling the load we should also measure latency.

I think it’s more important to deal with than the parallel decommission for now.

File: test-cases/performance/perf-regression-latency-650gb-grow-shrink.yaml [hunk 5]
```diff
@@ -1,21 +1,36 @@
...
 
-stress_cmd_w: "cassandra-stress write no-warmup cl=QUORUM duration=2850m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate 'threads=250 fixed=20332/s' -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(1..650000000,325000000,9750000)' "
-stress_cmd_r: "cassandra-stress read no-warmup  cl=QUORUM duration=800m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate 'threads=250 fixed=10310/s' -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(1..650000000,325000000,9750000)' "
-stress_cmd_m: "cassandra-stress mixed no-warmup cl=QUORUM duration=800m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate 'threads=250 fixed=8750/s' -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(1..650000000,325000000,6500000)' "
+# Enterprise cmd's
+stress_cmd_w: "cql-stress-cassandra-stress write no-warmup cl=QUORUM duration=2850m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate 'threads=250 fixed=21000/s' -col 'size=FIXED(128) n=FIXED(8)' -pop 'dist=gauss(1..650000000,325000000,9750000)' "
```
Comment (fruch) [inline test-cases/performance/perf-regression-latency-650gb-grow-shrink.yaml:12]: let's split to two case, one with c-s and one with cql-stress

File: sdcm/nemesis.py [hunk 6]
```diff
@@ -4034,6 +4090,16 @@ def _grow_cluster(self, rack=None):
...
 
+    def _grow_cluster_parallel(self, rack=None):
+        if rack is None:
+            rack = 0
+        add_nodes_number = self.tester.params.get('nemesis_add_node_cnt')
+        self.log.info("Start grow cluster on %s nodes", add_nodes_number)
+        InfoEvent(message=f"Start grow cluster on {add_nodes_number} nodes").publish()
+        self.add_new_nodes_parallel(count=add_nodes_number, rack=rack)
+        self.log.info("Finish cluster grow")
+        time.sleep(300)  # TODO: currently, just in case, to be removed
```
Comment (fruch) [inline sdcm/nemesis.py:4101]: Consider if we can remove

File: sdcm/nemesis.py [hunk 7]
```diff
@@ -5004,6 +5077,22 @@ def disrupt_disable_binary_gossip_execute_major_compaction(self):
             self.target_node.restart_scylla_server()
             raise
 
+    def _wait_for_tablets_balanced(self):
```
Comment (fruch) [inline sdcm/nemesis.py:5080]: I would put it into sdcm.utils

File: defaults/aws_config.yaml [hunk 8]
```diff
@@ -7,14 +7,13 @@ user_credentials_path: '~/.ssh/scylla-qa-ec2'
 instance_type_loader: 'c6i.xlarge'
 instance_type_monitor: 't3.large'
 # manual on creating loader AMI's: see docs/new_loader_ami.md
-ami_id_loader: 'scylla-qa-loader-ami-v20-ubuntu22'
+ami_id_loader: 'ami-07dc305835cc4ccbf'
```
Comment (fruch) [inline defaults/aws_config.yaml:10]: lets raise a task/issue for creating a new loader ami with tablets aware drivers

Comment (soyacz) [reply defaults/aws_config.yaml:10]: https://github.com/scylladb/scylla-cluster-tests/issues/7663

**Comment by fruch** [comment]
@soyacz 

we can close this one ?

**Comment by soyacz** [comment]
> @soyacz
> 
> we can close this one ?

yes

**Pull request closed without merging**